### PR TITLE
gdb: update to 10.2

### PR DESCRIPTION
**Pull request by neheb**
Replace some OpenWrt patches with openembedded ones for easier
maintainability. Remove several outdated ones as well.

Replace PKG_RELEASE with AUTORELEASE to avoid manual bumps.

Remove !arc dependency as there are openembedded patches that add
support now.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
This was build tested on arm64, ppc, and ARC. No issues.

**Comment by neheb** [comment]
removed ARC patches based on IRC feedback.

File: package/devel/gdb/Makefile [hunk 1]
```diff
@@ -45,7 +45,6 @@ endef
 define Package/gdbserver
 $(call Package/gdb/Default)
   TITLE:=Remote server for GNU Debugger
-  DEPENDS+=@!arc
```
Comment (hauke) [inline package/devel/gdb/Makefile:48]: Please do not remove the @!ARC

Comment (hauke) [reply package/devel/gdb/Makefile:48]: It compiles also on arc

**Comment by hauke** [comment]
Thank you for the patch, I applied it to master.

**Pull request closed without merging**